### DISCUSSION TITLE
Is there a way to use extensions that are blocked at the firewall level?

### DISCUSSION THREAD

**Discussion by jschwalbe** [coder/code-server#7005] [Q&A]
More specifically: I can access my code-server instance from home without issue. I tried to install the Codeium extension, but my work blocks everything AI at the firewall level. This means that while the extension is installed on the server, I cannot login to Codeium and therefore cannot do anything.

So is there a way to make all access to Codeium go through code-server (like a proxy) rather than through my work's internet?

Not sure if it is related, but I'm seeing this error:
```
code-server  | [20:07:17] [192.168.24.180][d4286449][ExtensionHostConnection] <1546> (VSCode) 2024-09-19 20:07:17 [ERROR]: connect ECONNREFUSED 127.0.0.1:39559
```

Thanks

**Comment by code-asher** [comment]
I do not think we can control this from code-server (the extension decides what URLs to open) but you could probably configure your browser to use an http proxy.

**Reply by jschwalbe** [reply]
OK thanks. I was hoping that code-server would essentially proxy communications from the extensions to whatever they need to connect to.
